File: templates/hexagonal-spring-boot/jpa-adapter/src/test/java/packageName/repository/ExampleJpaAdapterApplication.java
package packageName.repository;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.test.context.TestConfiguration;
import org.springframework.context.annotation.Bean;
import packageName.domain.port.ObtainExample;
import packageName.repository.dao.ExampleDao;

@SpringBootApplication
public class ExampleJpaAdapterApplication {

  public static void main(String[] args) {
    SpringApplication.run(ExampleJpaAdapterApplication.class, args);
  }

  @TestConfiguration
  static class ExampleJpaTestConfig {

    @Bean
    public ObtainExample getObtainExampleRepository(ExampleDao exampleDao) {
      return new ExampleRepository(exampleDao);
    }
  }
}

File: templates/hexagonal-spring-boot-kotlin/jpa-adapter/src/test/kotlin/packageName/repository/ExampleJpaAdapterApplication.kt
package packageName.repository

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.test.context.TestConfiguration
import org.springframework.context.annotation.Bean
import packageName.domain.port.ObtainExample
import packageName.repository.dao.ExampleDao

@SpringBootApplication
class ExampleJpaAdapterApplication {

    fun main(args: Array<String>) {
        SpringApplication.run(ExampleJpaAdapterApplication::class.java, *args)
    }

    @TestConfiguration
    class ExampleJpaTestConfig {
        @Bean
        fun getObtainExampleRepository(exampleDao: ExampleDao): ObtainExample {
            return ExampleRepository(exampleDao)
        }
    }
}
